### DISCUSSION TITLE
How can you use code annotation in a dynamically created code chunk in a string?

### DISCUSSION THREAD

**Discussion by liyongg** [quarto-dev/quarto-cli#6709] [Q&A]
### Description

I would like to use the code annotation feature with dynamically created code chunks in a string. In this example I try to create variables in a glue string dynamically. Aside from RStudio not being happy with handling this type of request in the visual/source editor, I can't seem to get the code annotation functioning. I used `output: asis` to render the output of the code as markdown, but it does not seem to go well with the annotations.

Context: I am writing documentation to showcase which lines can be copypasted in a `.qmd` file to plot certain figures and tables. The code should not be run and only be displayed as in the output picture.

Do you have any pointers or is the result I want to achieve not even possible?

## The input 

````qmd
---
title: "Untitled"
---

```{r}
#| output: asis
#| echo: false
library(glue)
library(dplyr)
var_mpg <- "mpg"
test <- glue(.open = '[', .close = ']',
             "```{{r}}
             mtcars |>
              mutate(var = [var_mpg] * [var_mpg], # <1>
                     var2 = [var_mpg] / [var_mpg]) # <2>
             ```
             1.  Please
             2.  Appear
             ")

cat(test)
             
```
````

## The output
![image](https://github.com/quarto-dev/quarto-cli/assets/44399190/15bf45cf-2bd2-4f2a-9f11-f1184c7e89c8)

## Session
```
R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)
Quarto: 1.4.346
RStudio: RStudio 2023.06.2+561 "Mountain Hydrangea" Release (de44a3118f7963972e24a78b7a1ad48b4be8a217, 2023-08-25) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.06.2+561 Chrome/110.0.5481.208 Electron/23.3.0 Safari/537.36
```

**Comment by cderv** [comment]
I believe you are just encountering the limitation that code annotation does not work yet on fenced code block

* https://github.com/quarto-dev/quarto-cli/issues/6385

**Reply by liyongg** [reply]
I had read #6244 before posting this and saw it as closed and assumed it would work, but I forgot to mention it in my initial post (my bad). It seems like I have misinterpreted the title of that issue? The other issue you linked does indeed look like the problem I have here (:

**Reply by cderv** [reply]
It is almost the same. Slight difference in how it behaves internally and what it outputs in .md due to the fact that using `echo: fenced` is applied on a computation code chunk. In your case, what is produced in .md is a normal code chunk. 

So #6385 is about detecting directly in .md file
````
```{{r}}
````
 while #6244 was about detecting same pattern, inside another code block.
